pkg/com/stat.go

```diff
@@ -0,0 +1,74 @@
+package com
+
+import (
+	"encoding/json"
+	"sync"
+	"time"
+)
+
+// jsonStat allows field access from json.Marshal().
+type jsonStat = struct {
+	// Data contains ops/s by second.
+	Data []uint64 `json:"data"`
+	// Mtime is the last timestamp of Data.
+	Mtime int64 `json:"mtime"`
+}
+
+// Stat records ops/s by second for computing ops/$SPAN.
+type Stat struct {
+	jsonStat
+
+	mtx sync.RWMutex
+}
+
+func NewStat(seconds uint) *Stat {
+	return &Stat{jsonStat: jsonStat{Data: make([]uint64, seconds)}}
+}
+
+// Record increments the ops of the current second by i.
+func (s *Stat) Record(i uint64) {
```

Comment:
You know I like tests and for testability it would be great to have the actual implementation in a function taking the current time as a parameter and a trivial wrapper that calls it with the current time.